src/osd/PrimaryLogPG.cc

```diff
@@ -5931,7 +5931,7 @@ int PrimaryLogPG::do_sparse_read(OpContext *ctx, OSDOp& osd_op) {
     }
 
     bufferlist data_bl;
-    r = pgbackend->objects_readv_sync(soid, std::move(m), op.flags, &data_bl);
+    r = pgbackend->objects_readv_sync(soid, m, op.flags, &data_bl);
```

Comment:
I think the problem is that `m` still can be used, even after it has been std::moved out here (see https://github.com/ceph/ceph/pull/54047/commits/2e87bc047b6337238e86e2e65eae9e0f46c8e1f5#diff-04d5a3324b706d64a3230221d2eb08f6aa66829f93ca151ebb71896fb743ada7R5962).

Reply:
Please see https://github.com/ceph/ceph/pull/54047#discussion_r1364766724, we should change it back to `&` anyway.